docker-compose.yaml

```diff
@@ -22,13 +27,53 @@ services:
     build: .
     volumes:
      - .:/dockermount
+    depends_on:
+      - usaspending-db
     command: python3 -u manage.py shell
     environment:
       DJANGO_DEBUG: ${DJANGO_DEBUG}
       DATABASE_URL: postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@${POSTGRES_HOST}:${POSTGRES_PORT}/data_store_api
       ES_HOSTNAME: ${ES_HOST}:${ES_PORT}
       DATA_BROKER_DATABASE_URL: postgresql://${BROKER_USER}:${BROKER_PASSWORD}@${BROKER_HOST}:${BROKER_PORT}/data_broker
 
+  usaspending-test:
+    build: .
+    volumes:
+     - .:/dockermount
+    depends_on:
+      - usaspending-db
+      - usaspending-es
+    command: pytest
```

Comment:
We may need to let this one sleep a few seconds on a totally fresh/clean setup if it truly depends on the DB and ES. The `depends_on` directive doesn't really have readiness checks built-in by default. 

There _was_ a cool way in docker-compose v2 to do this with `healthcheck` and `depends_on` with a healthy condition. See:  https://github.com/peter-evans/docker-compose-healthcheck ... 

But since v3 of docker-compose they are getting away from that (see note in the beginning of the above link). So you're left with either a rudimentary sleep, or one of the `wait-for-it.sh` scripts that are out there.

Reply:
It seems like an edge case which has a very straightforward and convenient workaround: try again. 

If a decision needs to be made between attempting to preemptively protect a first-time user or making it more efficient for repeated use I will lean toward the repeated use. A number of the changes in this PR are to make the docker-compose managed env _convenient_ for developers (backend and frontend) so there is more buy-in of the tool and it becomes the de facto local env across the team. Adding a small delay doesn't sound like much, but it would be unnecessary for over 95% of the usage and add a strike against a migration to the docker managed env.

I am certainly not arguing to ignore the needs of first-time contributors. The dockerization and compose services should be enough support for first-time contributors especially if the docker-compose script is widely used and maintained moving forward.